generators/server/templates/entity/src/main/java/package/service/mapper/EntityMapper.java.ejs

```diff
@@ -109,9 +109,17 @@ public interface <%= entityClass %>Mapper extends EntityMapper<<%= dtoClass %>,
     @Mapping(target = "<%= field.propertyName %>", source = "<%= field.propertyName %>")
     <%_ } _%>
     <%_ if (!relatedField.id) { _%>
+      <%_ if (relatedField.mapstructExpression) { _%>
```

Comment:
the expression is only used when there's an id? (line above)

Reply:
I think, it just generates the code if the relatedField is not an id field (so just for regular fields)